### PR TITLE
[ENH] isolate `statsmodels` imports, part 2

### PR DESCRIPTION
**Pull request by fkiraly**
Isolates remaining `statsmodels` imports, continues #3445 (see description there).

### CODE REVIEW

File: sktime/_contrib/datasets/_single_problem_loaders.py [hunk 1]
```diff
@@ -936,6 +936,9 @@ def load_macroeconomic():
     .. [3] Data Source: Bureau of Labor Statistics, U.S. Department of Labor;
           http://www.bls.gov/data/; accessed December 15, 2009.
     """
+    _check_soft_dependencies("statsmodels")
```
Comment (GuzalBulatova) [inline sktime/_contrib/datasets/_single_problem_loaders.py:939]: Shouldn't we introduce `_check_soft_dependencies("statsmodels")` before every statsmodels import then? In #3445 most (maybe all) of the updated files don't have that.

**Review by GuzalBulatova — COMMENTED** [review]
I don't like how these imports look, but I understand the why. Is this a statsmodels-specific thing?

File: sktime/_contrib/datasets/_single_problem_loaders.py (hunk 1 above)
Comment (fkiraly) [reply sktime/_contrib/datasets/_single_problem_loaders.py:939]: the trick is in the `python_dependencies` tag, which has the value `"statsmodels"`, in those cases.

If that is set, the check is called by the base class (which reads the tag). Single locus of responsibility, for lookup as well as dependency checking.

Comment (fkiraly) [reply sktime/_contrib/datasets/_single_problem_loaders.py:939]: > Is this a statsmodels-specific thing?

No, generally we want ot isolate soft dependencies to inside classes or functions. Isolating statsmodels is a "good thing" imo, see #3445, even if it remains a core dependency.

Comment (GuzalBulatova) [reply sktime/_contrib/datasets/_single_problem_loaders.py:939]: Ok, I understand, thanks for clarification, Franz

**Pull request merged**